libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/OAuth2.java

```diff
@@ -441,6 +440,28 @@ private static TokenEndpointResponse makeTokenEndpointRequest(HttpAccess httpAcc
         }
     }
 
+    /**
+     * Fetches an OpenID token from the Salesforce backend. This requires an OpenID token to be
+     * configured on the Salesforce connected app in the backend. It also requires the "openid"
+     * scope to be added on the client side through bootconfig and on the connected app.
+     *
+     * @param loginServer Login server.
+     * @param clientId Client ID.
+     * @param refreshToken Refresh token.
+     * @return OpenID token.
+     */
+    public static String getOpenIDToken(String loginServer, String clientId, String refreshToken) {
+        String idToken = null;
+        try {
+            final TokenEndpointResponse tr = refreshAuthToken(HttpAccess.DEFAULT,
```

Comment:
Currently, we have to use the `/token` endpoint to fetch OpenID token, but we might have a dedicated endpoint in the future. At such a time, we can swap out the implementation here to use that new endpoint, keeping the Mobile SDK API the same.